Various minor fixes of Library and Upload page

pva701 opened the pull request:
## Description

<!--
Describes the nature of your changes. If they are substantial, you should
further subdivide this into a section describing the problem you are solving and
another describing your solution.
-->

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
-->

https://issues.serokell.io/issue/EDNA-93

## :white_check_mark: Checklist for your Pull Request

<!--
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
-->

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

pva701 committed:
Add path fragments for Library

Problem: on refersh current tab in library is reset.

Solution: make tabs path fragments.

pva701 committed:
Make View buttons on Upload page work

Problem: they didn't work.

Solution: make them preselect project, compound and target on Dashboard page.

gromakovsky commented:
I see that it does two quite unrelated things that belong to two different issues. Can you please split this PR into 2? Also there is no description.

pva701 commented:
>Can you please split this PR into 2

No, have no sense to follow these stupid rules. The issues are minor.

pva701 commented:
Stop being robot who follows the predefined algorithms and rules, let's abandon these unbearable rules at least for such cases as this PR.

pva701 commented:
>Also there is no description.

stop this bullshit, stop please, I don't want to hear this. You just follow some predefined commands. Don't you really understand what it's about? you perfectly know, you know even better than me. Stop this useless activity.

Don't even tell me that this is open-source repo and someone else can read this and be interested. No, it's not even close to the reality. When this repo will be at least slightly resembles open source projects with interested parties and devs apart from us, I will follow these rules. Stop making statements that are completely untrue and force me doing some useless activity relying on these statements.  

Moreover, there are commit messages which I wrote exactly not to hear this shit, though even they are totally useless, nobody will ever read them apart from main titles. So if you fell down and lost you memory and forgot what these issues are, you can freely read commit messages and figure things out.

pva701 commented:
There is no any practical sense in doing so. Just to satisfy your PROPERNESS, but asking me do so, firstly waste my mental health maybe you don't care but due to this I just don't want to touch this project and contact with you at all, secondly it takes time to write these comments to reason this bullshit at 99th time, thirdly it will take time to rebase to split this into two, why, why I should spend time on this when I can do something useful, some real tasks? You just don't understand: working day, week, deadline has LIMITED, LIMITED amount of time and compelling me doing useless tasks, which would take just 20 minutes, I spend 20 minutes of time which could be spend on something which has real impact on the results of the project. I just got tired of explaining this, I just got tired of how you keep repeating such useless things. You waste my, waste your, waste project time on this. I waste time on replying to you. I don't want to waste even more time rebasing this shit, I just hate rebasing and I just hate how useless rebasing is especially when it's a mandatory case but just your whim.

pva701 commented:
And what is YT task? Task is created by you or by me, it might have arbitrary scope, why is it source of truth (especially in fixes case)? why not to make all small changes in the one task. You don't even review them to say me: you know it's hard to review, and it's not truth. You just asking me to follow these idiotic rules without realising why to do so. Maybe we will create a PR for every commit? 

For me tasks are either independent piece of new functionality, which is logical to make separately and review separately, or it's banch of fixes. Why not to have them in all task PR? They are context free, they are independent (so they don't impose a lot of context for reviewer, like before in this PR his made this this and this, and this commit adds something). They are easy to review, they are easy to NOT HAVE CONFLICTS. There is no practical sense in having this tiny PRs for each tiny task which you came up with. Like you want some small fix, you create a task. You want one more, you create another. Ok, it makes sense, not to forget about these fixes. But why it makes sense to have separate PRs? No sense. Once again: you don't even review PRs to complain that it's hard to review. You just came and starting this whining: "You know guys we have too much time to prepare decent MVP, let's spend this time somehow. It's so pleasant to have big tracks rebasing for me, so I want you to have big tracks as well. It's needed for nothing, we will forget about this immediately after we close this tiny PR, but let's just waste time and follow these stupid rules which we introduced and don't even ever doubt them and don't ask ourselves, if it makes sense to do so or not. Just let's waste time regardless of sanity"

pva701 committed:
Notify about unsaved data on leaving Upload page

Problem: currently if user tries to leave Upload page with some filled fields,
he will lose them.

Solution: add alert to warn user about it.

pva701 committed:
Make standard form elements to inherit font-family from containers

Problem: standard elements like button, input, textarea
don't inherit our default font from body by default,
so we get discrepancy witht the design.

Solution: make them inherit, also revert font-weight: 700 for buttons.

sashasashasasha151 commented on frontend/src/components/Button/Button.tsx:
```diff
@@ -47,7 +49,9 @@ export function Button({
           onClick();
         }
-        e.stopPropagation();
+        if (!propagate) {
+          e.stopPropagation();
+        }
```
Maybe I missed smth but what for you need `stopPropagation` here and what for you don't need it?

sashasashasasha151 replied:
So, I removed it and looks like all works fine

pva701 replied:
I remembered: without it if we stop click propagation underlying `<Link>` not passes `state: { projectId, compoundId: p.compoundId, targetId: p.targetId },`

sashasashasasha151 commented on frontend/src/components/UploadPreviewTable/UploadPreviewTable.tsx:
```diff
@@ -38,17 +40,27 @@ export function UploadPreviewTable({
+              }}
+            >
+              <Button
+                propagate
+                type="link"
+                disabled={!projectId || !isDefined(p.compoundId) || !isDefined(p.targetId)}
```
Is it possible

pva701 replied:
it's, if compound or target are new

pva701 replied:
also, I pass
`projectId={excelFile.state === "added" ? currentProject?.id : undefined}`
in `UploadPage.tsx`, so until a user added an experiment I pretend it doesn't have project (kostil)

sashasashasasha151 commented on frontend/src/components/UploadPreviewTable/UploadPreviewTable.tsx:
```diff
@@ -38,17 +40,27 @@ export function UploadPreviewTable({
-            <Button type="link" disabled={!viewEnabled}>
-              View
-            </Button>
+            <Link
```
![image](https://user-images.githubusercontent.com/25207759/115850992-2371a280-a42f-11eb-8fbc-1ad9eaf70ff2.png)
Looks like I can click on disabled `view`

pva701 replied:
good catch

sashasashasasha151 replied:
![image](https://user-images.githubusercontent.com/25207759/116228773-6482f300-a75e-11eb-8be6-2a5cc6f8768a.png)
Looks like now view doesn't work

sashasashasasha151 commented on frontend/src/pages/library/LibraryPage.scss:
```diff
@@ -18,8 +18,9 @@
     font-size: 26 * $px;
     font-weight: 700;
-    line-height: 32 * $px;
+    line-height: 36 * $px;
```
Why?

pva701 replied:
to make it the same height as `+ project/+ methodology` buttons to avoid jumping when you switch between tabs

sashasashasasha151 commented on frontend/src/pages/library/LibraryPage.tsx:
```diff
@@ -57,24 +63,45 @@ interface EntitiesTabProps {
       </div>
-      {render(entityTab)}
+
+      <Switch>
+        {entitiesFlds.map(entity => renderRoute(entity))}
+
+        <Redirect exact from="/library" to="/library/projects" />
+
+        <Route path="*">
+          <NotFound />
+        </Route>
+      </Switch>
```
Isn't it possible to put it to all other routers? Or it is known pattern?

pva701 replied:
I've seen such patterns in react-router tutorial, where Switch is used not only as global router mechanism but some local one.

I don't know if it's known pattern or not, but why should we create this Library implementation details implementation? Considering this, I decided to put it here instead of putting directly in App. I

gromakovsky commented:
@pva701 
<img width="551" alt="image" src="https://github.com/serokell/edna/assets/2556409/7d8bc727-94f7-4097-8322-8cf42cec63e7">

pva701 commented:
@gromakovsky yes this is about me. I am fluttered you remember this PR, hopefully, you store printed copy of it on your desk framed ❤️
